### PR TITLE
Creating login & logout features

### LINKED ISSUE #78: Create login feature
Create an ability for registered users to log in to the website.
- Create a template for login.
- Create a login view.
- Create a login route.
Required outcome:
A user should be able to login and be redirected to the homepage. Plus, a user should be notified if something is wrong with his credentials.

### PR DESCRIPTION
**Pull request by DanielMy777**
- Creating login & logout templates (using crispy forms).
- Creating url's for the login & logout pages in main/urls.py.
- Connecting landing page redirect to the login page when clicking 'login'.
- Adding tests to login & logout features.
NOTE: The login form is the default that Django provides us. (LoginView)
[THIS PR IS DEPENDANT ON #94 - needs a rebase after the merge  - DONE]
fix #78 
image

### CODE REVIEW

**Comment by DanielMy777** [comment]
Note that 2 commits will drop after the merge of #94 and shrink this PR

**Comment by nadavsu** [comment]
Nice work!

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Please rebase.

**Comment by DanielMy777** [comment]
> Please rebase.

@Omeramsc
I rebased the PR it is now ready for review.

**Comment by tamirho** [comment]
Looks really good!! Nice work!

File: main/urls.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 from django.urls import path
+from django.contrib.auth import views as auth_views
 from . import views
 
 urlpatterns = [
     path('', views.home, name='home'),
+    path('login/', auth_views.LoginView.as_view(template_name='main/login.html'), name='login'),
+    path('logout/', auth_views.LogoutView.as_view(template_name='main/logout.html'), name='logout'),
```
Comment (simzacks) [inline main/urls.py:8]: Is this actually doing login and logout? Or is it just opening html pages for now saying that it is done?

Comment (DanielMy777) [reply main/urls.py:8]: @simzacks It's logging in and out - fully functional.

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Please add tests for the features.

**Comment by DanielMy777** [comment]
> Please add tests for the features.

@Omeramsc 
The tests have been added to this PR, please review again! :)

**Review by simzacks — APPROVED** [review]

**Review by OrAsaf — APPROVED** [review]

**Pull request merged**